src/Stratis.SmartContracts.Executor.Reflection/CallDataSerializer.cs

```diff
@@ -13,13 +14,13 @@ public class CallDataSerializer : ICallDataSerializer
         // TODO this is ugly but there is poor DI support for rules so we can't inject it yet
         public static ICallDataSerializer Default = new CallDataSerializer(new MethodParameterSerializer());
 
-        private readonly IMethodParameterSerializer methodParamSerializer;
+        public IMethodParameterSerializer MethodParamSerializer { get; }
```

Comment:
This feels a little lazy

Reply:
It is. It imagined it could be fixed/removed in part 2, when the param serialization is changed.